tests/blackbox/stratis_cli_cert.py

```diff
@@ -20,12 +20,77 @@
 import time
 import unittest
 
-from testlib.utils import exec_command, exec_test_command, process_exists
+from testlib.utils import exec_command, exec_test_command, process_exists, p_n, fs_n
 from testlib.stratis import STRATIS_CLI, StratisCli, clean_up
 
 DISKS = []
 
 
+def make_test_pool():
+    """
+    Create a test pool that will later get destroyed
+    :return: Name of the created pool
+    """
+    pool_name = p_n()
+    exec_test_command([STRATIS_CLI, "pool", "create", pool_name, DISKS[0]])
+    return pool_name
+
+
+def make_test_filesystem(pool_name):
+    """
+    Create a test filesystem that will later get destroyed
+    :param pool_name: Name of a test pool
+    :return: Name of the created filesystem
+    """
+    filesystem_name = fs_n()
+    assert (
+        exec_test_command(
+            [STRATIS_CLI, "filesystem", "create", pool_name, filesystem_name]
+        )[0]
+        == 0
+    )
+    return filesystem_name
+
+
+def assert_test_result_success_stdout_empty(self, exit_code, stdout, stderr):
```

Comment:
Two comments on these abstracted testing methods:

1. It probably makes sense to pass the command line as an argument also.

2. You can probably collapse these methods all into a single method more efficiently. I think originally I thought there were only 8 possibilities, but actually there are 12. We have 3 possibilities for the error code: 0 (when things go well), 2 (when there is an error in parsing the command line arguments), 1 (for all other errors). For stdout and stderr we're trying to go w/ distinguishing only between empty and non-empty. This gives 3 * 2 * 2 = 12 possible choices. We don't want 12 methods distinguished by their names, so it's probably better to have one method w/ parameters that govern its behavior.

In your scheme, you are already passing the expected exit code. This means that you would expect to have 12 / 3 = 4 distinct methods. But it would really be easier and more extensible for the user if you used, for example, boolean arguments to indicate whether stderr and stdout were expected to be empty or non-empty.